diff -ur -x '*.o' -x Makefile -x lgeneral -x .deps -x .dirstamp lgeneral-orig/src/slot.c lgeneral-1.4.4/src/slot.c
--- lgeneral-orig/src/slot.c	2021-01-06 09:28:42.000000000 +0100
+++ lgeneral-1.4.4/src/slot.c	2026-09-16 06:37:43.403210829 +0200
@@ -226,7 +226,7 @@
 ====================================================================
 */
 enum UnitStoreParams {
-    UnitLibSize = 4*sizeof(void *) + 26*sizeof(int) + TARGET_TYPE_LIMIT*sizeof(int)
+    UnitLibSize = 4*sizeof(void *) + 27*sizeof(int) + TARGET_TYPE_LIMIT*sizeof(int)
 #ifdef WITH_SOUND
             + 1*sizeof(void *) + 1*sizeof(int)
 #endif
diff -ur -x '*.o' -x Makefile -x lgeneral -x .deps -x .dirstamp lgeneral-orig/src/unit_lib.c lgeneral-1.4.4/src/unit_lib.c
--- lgeneral-orig/src/unit_lib.c	2020-10-31 11:02:31.000000000 +0100
+++ lgeneral-1.4.4/src/unit_lib.c	2026-09-16 12:42:40.919577749 +0200
@@ -446,6 +446,7 @@
          * if picture_type is not ALL_DIRS, picture is a single picture looking to the right;
          * add a flipped picture looking to the left 
          */
+		unit->icon_o = offset;
         if ( unit->icon_type == UNIT_ICON_ALL_DIRS ) {
             unit->icon = create_surf( width * 6, height, SDL_SWSURFACE );
             unit->icon_w = width;
@@ -571,6 +572,14 @@
                 unit->eval_score /= best_ground;
         }
     }
+#if 1
+	list_reset(unit_lib);
+	FILE* jsf = fopen("/tmp/icons.js", "w");
+    while (unit = list_next(unit_lib)) {
+		fprintf(jsf, "{o:%d,w:%d,h:%d},\n", unit->icon_o, unit->icon_w, unit->icon_h);
+	}
+	fclose(jsf);
+#endif
     free(domain);
     SDL_FreeSurface(icons);
     return 1;
diff -ur -x '*.o' -x Makefile -x lgeneral -x .deps -x .dirstamp lgeneral-orig/src/unit_lib.h lgeneral-1.4.4/src/unit_lib.h
--- lgeneral-orig/src/unit_lib.h	2020-10-31 11:02:32.000000000 +0100
+++ lgeneral-1.4.4/src/unit_lib.h	2026-09-16 06:37:43.403729284 +0200
@@ -145,6 +145,7 @@
     SDL_Surface *icon_tiny; /* half the size; used to display air and ground unit at one tile */
     int icon_type;          /* either single or all_dirs */
     int icon_w, icon_h;     /* single icon size */
+    int icon_o;
     int icon_tiny_w, icon_tiny_h; /* single icon size */
     int flags;
     int start_year, start_month, last_year; /* time of usage */
